packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -222,22 +242,38 @@ module.exports = {
                 },
                 {
                   loader: require.resolve('postcss-loader'),
+                  options: postCSSLoaderOptions,
+                },
+              ],
+            },
+            extractTextPluginOptions
+          )
+        ),
+        // Note: this won't work without `new ExtractTextPlugin()` in `plugins`.
+      },
+      // Adds support for CSS Modules (https://github.com/css-modules/css-modules)
+      // using the extension .module.css
+      {
+        test: /\.module\.css$/,
+        loader: ExtractTextPlugin.extract(
+          Object.assign(
+            {
+              fallback: require.resolve('style-loader'),
+              use: [
+                {
+                  loader: require.resolve('css-loader'),
                   options: {
-                    ident: 'postcss', // https://webpack.js.org/guides/migrating/#complex-options
-                    plugins: () => [
-                      require('postcss-flexbugs-fixes'),
-                      autoprefixer({
-                        browsers: [
-                          '>1%',
-                          'last 4 versions',
-                          'Firefox ESR',
-                          'not ie < 9', // React doesn't support IE8 anyway
-                        ],
-                        flexbox: 'no-2009',
-                      }),
-                    ],
+                    importLoaders: 1,
+                    minimize: true,
+                    sourceMap: true,
+                    modules: true,
+                    localIdentName: '[name]__[local]___[hash:base64:5]',
```

Comment:
Names could get really long then, no? Especially if you import a `.module.css` file from `node_modules` somewhere (is CRA supporting that? I really hope so).

Stable names across multiple builds would be nice though. Would be great if the path string itself could be hashed, but I don't think that's an option.

Forgive me for my lack of context (I haven't been following super closely but am keenly interested in inclusion of css-modules support), but are we on webpack 2 at this point? It seems like css-loader supports passing a custom `getLocalIdent` function as an option, but that only works under webpack >= 2.2.1 (https://github.com/webpack-contrib/css-loader#css-scope)